slasher/detection/attestations/spanner.go

```diff
@@ -219,13 +234,18 @@ func (s *SpanDetector) updateMinSpan(ctx context.Context, att *ethpb.IndexedAtte
 				indices = append(indices, idx)
 			}
 		}
-		if err := s.slasherDB.SaveEpochSpansMap(ctx, epoch, spanMap); err != nil {
-			return err
-		}
-		if len(indices) == 0 {
-			break
+		//copy(valIndices, indices)
+		if !batchRead {
```

Comment:
There's a lot of complex !batchRead conditionals around this function - can we refactor for easier reading and understanding of the conditionals?

Reply:
sure